core/schema/file.graphqls

```diff
@@ -18,4 +18,7 @@ type File {
 
     "The size of the file, in bytes"
     size: Int!
+
+    "Write the file to a directory on the host"
+    export(path: String!): Boolean!
```

Comment:
I get why we want to make `path` be a directory on the host (it's annoying to have to specify the filename all the time), but my initial intuition would have been that `path` should be set to the path where I want the *file* at, not the parent dir.

The problem is that if a user has that expectation and does something like export a file named "foo.txt" like this:
```go
file.Export(ctx, "/dir/subdir/bar.txt")
```

The actual result will be that we create `/dir/subdir/bar.txt/foo.txt` even though they probably expected us to create `/dir/subdir/bar.txt` with the contents of `foo.txt`.

I don't have great solutions in the short term for this other than considering making it so `path` points to the path of the file rather than the parent dir. But I think at minimum we could be more explicit in the docstring here.

Reply:
Agree. I actually started with 'file or dir' but then realized there's no way for us to tell the user's intention (besides requiring a trailing `/` which is too subtle).

So I went with just directory, but now that you mention it exporting to a file path feels much more intuitive alongside the container export.

I think it can be done easily by specifying a new name for the file on the right-hand side of `llb.Copy` and just exporting it to the specified file path's parent dir.

Starting on that now!

Reply:
p.s.: the only downside to this is kind of the opposite of what you mention: if someone does `file.Export(ctx, ".")` (targeting a dir) it won't work. I would expect some kind of 'is a directory' error.

Reply:
Oh, yikes, the Buildkit exporter actually seems to do a `os.RemoveAll` of the destination directory in order to replace it with a file. So that actually "works" but doesn't do NEARLY what the user wanted. :scream: 

I'll add a guard for this. Yet another thing to keep in mind with remote filesync. :thinking: